scipy/_lib/deprecation.py

```diff
@@ -110,3 +111,61 @@ def deprecate_cython_api(module, routine_name, new_name=None, message=None):
     if not has_fused:
         d[_DeprecationHelperStr(routine_name, depdoc)] = d.pop(routine_name)
 
+
+# taken from scikit-learn, see
+# https://github.com/scikit-learn/scikit-learn/blob/1.3.0/sklearn/utils/validation.py#L38
+def _deprecate_positional_args(func=None, *, version="1.14"):
```

Comment:
Why do we need the defaults here? Having a default version is going to get out of date quickly